plutus-ir/test/Spec.hs

```diff
@@ -57,163 +56,51 @@ compileAndMaybeTypecheck doTypecheck pir = flip runReaderT NoProvenance $ runQuo
     pure compiled
 
 tests :: TestNested
-tests = testGroup "plutus-ir" <$> sequence [
-    prettyprinting,
-    parsing,
-    datatypes,
-    recursion,
-    serialization,
-    errors,
-    optimizer,
-    transform
+tests = testGroup "plutus-ir" <$> sequence
+    [ prettyprinting
+    , parsing
+    , datatypes
+    , recursion
+    , serialization
+    , errors
+    , optimizer
+    , transform
     ]
 
 prettyprinting :: TestNested
-prettyprinting = testNested "prettyprinting" [
-    goldenPir "basic" basic
-    , goldenPir "maybe" maybePir
+prettyprinting = testNested "prettyprinting"
+    $ map (goldenPir id term)
+    [ "basic"
+    , "maybe"
     ]
 
-basic :: Term TyName Name ()
-basic = runQuote $ do
-    a <- freshTyName () "a"
-    x <- freshName () "x"
-    pure $
-        TyAbs () a (Type ()) $
-        LamAbs () x (TyVar () a) $
-        Var () x
-
-maybePir :: Term TyName Name ()
-maybePir = runQuote $ do
-    mb@(Datatype _ _ _ _ [_, just]) <- maybeDatatype
-    let unitval = embedIntoIR Unit.unitval
-    pure $
-        Let ()
-            NonRec
-            [
-                DatatypeBind () mb
-            ] $
-        Apply () (TyInst () (mkVar () just) Unit.unit) unitval
-
-listMatch :: Term TyName Name ()
-listMatch = runQuote $ do
-    lb@(Datatype _ l _ match [nil, _]) <- listDatatype
-
-    let unitval = embedIntoIR Unit.unitval
-
-    h <- freshName () "head"
-    t <- freshName () "tail"
-
-    let unitMatch = TyInst () (Var () match) Unit.unit
-    let unitNil = TyInst () (mkVar () nil) Unit.unit
-
-    pure $
-        Let ()
-            Rec
-            [
-                DatatypeBind () lb
-            ] $
-            mkIterApp () (TyInst () (Apply () unitMatch unitNil) Unit.unit)
-                [
-                    -- nil case
-                    unitval,
-                    -- cons case
-                    mkIterLamAbs () [VarDecl () h Unit.unit, VarDecl () t (TyApp () (mkTyVar () l) Unit.unit)] $ Var () h
-                ]
-
 datatypes :: TestNested
-datatypes = testNested "datatypes" [
-    goldenPlc "maybe" maybePir,
-    goldenPlc "listMatch" listMatch,
-    goldenEval "listMatchEval" [listMatch]
+datatypes = testNested "datatypes"
+    [ goldenPlcFromPir term "maybe"
+    , goldenPlcFromPir term "listMatch"
+    , goldenEvalPir term "listMatchEval"
     ]
 
 recursion :: TestNested
-recursion = testNested "recursion" [
-    goldenPlc "even3" evenOdd,
-    goldenEval "even3Eval" [evenOdd],
-    goldenPlc "mutuallyRecursiveValues" mutuallyRecursiveValues
+recursion = testNested "recursion"
+    [ goldenPlcFromPir term "even3"
+    , goldenEvalPir term "even3Eval"
+    , goldenPlcFromPir term "mutuallyRecursiveValues"
     ]
 
-natToBool :: Type TyName ()
-natToBool =
-    let nat = _recursiveType Nat.natData
-    in TyFun () nat Bool.bool
-
-evenOdd :: Term TyName Name ()
-evenOdd = runQuote $ do
-    let true = embedIntoIR Bool.true
```

Comment:
Agreed - the new `serializeEvenOdd` is quite bad because it's basically pure PLC, since it gets the definition of naturals, etc. from the PLC standard library. I've pushed a new one that uses the pir `datatype` definitions to simplify it a lot, it's actually readable now.